test/gtest/gpu_mha_ck_forward.cpp

```diff
@@ -0,0 +1,315 @@
+            std::get<tensor<T>>(tensors[miopenTensorMhaK]->m_cpu_tensor),
+            std::get<tensor<T>>(tensors[miopenTensorMhaV]->m_cpu_tensor),
+            softmax_ref,
+            mDesc_ref,
+            zInvDesc_ref,
+            q.mDescale,
+            k.mDescale,
+            v.mDescale,
+            s_descale,
+            s_scale,
+            o_scale,
+            drop,
+            std::get<tensor<int64_t>>(tensors[miopenTensorMhaDropoutSeed]->m_cpu_tensor)
+                .data.front(),
+            std::get<tensor<int64_t>>(tensors[miopenTensorMhaDropoutOffset]->m_cpu_tensor)
+                .data.front(),
+            amaxS_ref,
+            amaxO_ref,
+            oDesc_ref);
+    }
+
+    void TestBody() override
+    {
+        Handle& handle = get_handle();
+
+        std::vector<miopenSolution_t> solutions(16);
+        std::size_t found;
+
+        ASSERT_EQ(miopenFindSolutions(
+                      &handle, problem, nullptr, solutions.data(), &found, solutions.size()),
+                  miopenStatusSuccess);
+        ASSERT_GT(found, 0);
+        solutions.resize(found);
+
+        size_t workspace_size = 0;
+        Workspace workspace;
+
+        for(const auto& solution : solutions)
+        {
+            miopenGetSolutionWorkspaceSize(solution, &workspace_size);
+            workspace.resize(workspace_size);
+
+            ASSERT_EQ(
+                miopenRunSolution(
+                    &handle, solution, args.size(), args.data(), workspace.ptr(), workspace.size()),
+                miopenStatusSuccess);
+
+            auto GetResult = [this, &handle](miopenTensorArgumentId_t id, auto type) {
+                using ResultT         = std::decay_t<decltype(type)>;
+                auto& tensorStructPtr = tensors[id];
+                auto& cpu_tensor      = std::get<tensor<ResultT>>(tensorStructPtr->m_cpu_tensor);
+
+                cpu_tensor.data =
+                    handle.Read<ResultT>(tensorStructPtr->m_gpu_buffer, cpu_tensor.data.size());
+
+                return cpu_tensor;
+            };
+
+            auto gpu_tensor = GetResult(miopenTensorMhaO, T{});
+            double O_error  = miopen::rms_range(oDesc_ref, gpu_tensor);
+            // ck fp8 solver and low threshold
+            EXPECT_LT(O_error, ck_fp8_solver_threshold);
+        }
+    }
+
+    void TearDown() override
+    {
+        if(problem)
+        {
+            ASSERT_EQ(miopenDestroyProblem(problem), miopenStatusSuccess);
+        }
+    }
+
+    std::map<miopenTensorArgumentId_t, std::unique_ptr<TensorStruct>> tensors;
+    std::vector<miopenTensorDescriptor_t> descVector;
+    std::vector<miopenTensorArgument_t> args;
+
+    // ref data
+    tensor<float> softmax_ref;
+    tensor<T> oDesc_ref;
+    tensor<float> mDesc_ref;
+    tensor<float> zInvDesc_ref;
+    float amaxS_ref;
+    float amaxO_ref;
+
+    bool is_ck_solver             = false;
+    float ck_fp8_solver_threshold = 0.015;
```

Comment:
This is too high threshold, MIOpen fp8 algorithm fits into `0.0002` which is 75 times smaller.
Could you double-check that the values are similar, because rms error is not sensible to permutations and some significant outliers.